gateway/handlers/forwarding_proxy.go

```diff
@@ -131,7 +143,12 @@ func getServiceName(urlValue string) string {
 	var serviceName string
 	forward := "/function/"
 	if strings.HasPrefix(urlValue, forward) {
-		serviceName = urlValue[len(forward):]
+		r, _ := regexp.Compile(forward + "([^/?]+).*")
+		matches := r.FindStringSubmatch(urlValue)
+		// The matched subgroup will be the function name.
+		if 2 == len(matches) {
```

Comment:
What does the magic number 2 mean? What is the purpose of the regex?

Reply:
Similar to the above on adding back some comments...

So in this case, we want to figure out he function name from the URL.  Previously that was done by trimming off the `forward` length and taking the rest.  With a case like: `/function/xyz/my/path` though, what we really want is the section just following `/function/` up to, but not including any later path segments or query string.

We've got one regex subgroup to match, so per the `FindStringSubmatch` if there is a match, the returned slice will have the text of the leftmost match at `matches[0]` (in this case, `urlValue`) followed by the matched subgroups.  We've only got one subgroup, so what we want is `matches[1]`.  So what we want to see for a positive match is a `matches` where the length is exactly `2`.

I'll try to come up with a briefer way to say all that for the comment.